src/status_im2/contexts/syncing/scan_sync_code/view.cljs

```diff
@@ -255,136 +252,102 @@
+        qr-view-finder (reagent/atom {})
+        render-camera? (reagent/atom false)
+        torch?         (reagent/atom false)]
+    (fn [{:keys [title show-bottom-view? background animated? qr-code-succeed?
+                 set-qr-code-succeeded]}]
+      (let [torch-mode              (if @torch? :on :off)
+            flashlight-icon         (if @torch? :i/flashlight-on :i/flashlight-off)
+            scan-qr-code-tab?       (= @active-tab 1)
+            show-camera?            (and scan-qr-code-tab?
+                                         @camera-permission-granted?
+                                         @preflight-check-passed?
+                                         (boolean (not-empty @qr-view-finder)))
+            camera-ready-to-scan?   (and (or (not animated?) @render-camera?)
+                                         show-camera?
+                                         (not qr-code-succeed?))
+            title-opacity           (reanimated/use-shared-value (if animated? 0 1))
+            subtitle-opacity        (reanimated/use-shared-value (if animated? 0 1))
+            content-opacity         (reanimated/use-shared-value (if animated? 0 1))
+            content-translate-y     (reanimated/interpolate subtitle-opacity [0 1] [85 0])
             bottom-view-translate-y (reanimated/use-shared-value
                                      (if animated? (+ 42.2 (:bottom insets)) 0))
-            reset-animations-fn
-            (fn []
-              (reset! render-camera? false)
-              (js/setTimeout
-               (fn []
-                 (rf/dispatch [:navigate-back])
-                 (when @dismiss-animations
-                   (@dismiss-animations))
-                 (reanimated/animate-shared-value-with-timing
-                  content-opacity
-                  0
-                  (/ constants/onboarding-modal-animation-duration 8)
-                  :easing4)
-                 (reanimated/animate-shared-value-with-timing
-                  subtitle-opacity
-                  0
-                  (- constants/onboarding-modal-animation-duration
-                     constants/onboarding-modal-animation-delay)
-                  :easing4)
-                 (reanimated/animate-shared-value-with-timing title-opacity
-                                                              0
-                                                              0
-                                                              :easing4))
-               (if show-camera? 500 0)))]
-        (rn/use-effect (fn []
-                         (reset! app-state-listener
-                           (.addEventListener rn/app-state
-                                              "change"
-                                              #(when (and (not= % "active") @torch?)
-                                                 (reset! torch? false))))
-                         #(.remove @app-state-listener)))
```

Comment:
About the `app-state-listener` atom removal:

it's not required to be an atom, just a `let` binding:

```clojure
(let [res (.addEventListener something "something else" fn-callback)]
  #(.remove res))
```